sqlglot/dialects/snowflake.py

```diff
@@ -562,6 +562,14 @@ class Generator(generator.Generator):
             exp.VolatileProperty: exp.Properties.Location.UNSUPPORTED,
         }
 
+        def trycast_sql(self, expression: exp.TryCast) -> str:
+            value = expression.this
+            if value.is_string or value.is_type(*exp.DataType.TEXT_TYPES):
```

Comment:
what happens if the user does

parse_one("""
select TRY_CAST(x AS int)
""", read="snowflake")

x is a string, but the type isn't known to sqlglot, it should maintain a try_cast shouldn't it

Reply:
that's a good point thanks, will fix